pkg/controller/volume/cache/desired_state_of_world.go

```diff
@@ -57,7 +57,7 @@ type DesiredStateOfWorld interface {
 	// should be attached to the specified node, the volume is implicitly added.
 	// If no node with the name nodeName exists in list of nodes managed by the
 	// attach/detach attached controller, an error is returned.
-	AddPod(podName types.UniquePodName, volumeSpec *volume.Spec, nodeName string) (api.UniqueVolumeName, error)
+	AddPod(pod *api.Pod, volumeSpec *volume.Spec, nodeName string) (api.UniqueVolumeName, error)
```

Comment:
Although pod obj is big, but we are passing the pointer, will it still cause more overhead? I am also thinking that if later we need more information about the pod,  we have to keep changing the AddPod() interface.